src/tir/transforms/plan_update_buffer_allocation_location.cc

```diff
@@ -32,21 +32,21 @@
 namespace tvm {
 namespace tir {
 
-class CollectUnmanagedAllocations : public StmtExprVisitor {
+class CollectManagedAllocations : public StmtExprVisitor {
```

Comment:
Good point.  This was a bug that I encountered and fixed during implementation.  The TIR build pipeline made some simplifying assumptions about the TIR that it would receive, which the new legalization functions do not follow.

I've spun it out into an independent [PR#16726](https://github.com/apache/tvm/pull/16726).